README.md

```diff
@@ -1,57 +1,63 @@
+openssl req -nodes -x509 -days 365 -newkey rsa:2048 -keyout private.pem -out public.crt
 ```
 
-Copy the certificate to `/config/certs/sp/myapp.crt` in the
-[identity-idp](https://github.com/18F/identity-idp). Then, add the following
-entry to `config/service_providers.yml`:
-
-```yaml
-'urn:gov:gsa:openidconnect:sp:myapp':
-  agency: 'GSA'
-  cert: 'myapp'
-  friendly_name: 'Demo OmniAuth app'
-  logo: '18f.svg'
-  redirect_uris:
-    - 'http://localhost:4040/auth/logindotgov/callback'
-```
+Note: the `-nodes` flag skips encryption of the private key and is not recommended for production use.
 
-Run `rake db:seed` to update the service provider table.
+#### Register your app with the login.gov sandbox
 
-Now, the key can be added to your repo. For this example let's put it at
-`config/myapp.pem`.
+You will need to register your service provider with our sandbox environment using the login.gov [Partner Dashboard](https://dashboard.int.identitysandbox.gov). Note: this is not your production login.gov account.
 
-Add this gem to the Gemfile and run `bundle install`:
+1. Log in to the [Partner Dashboard](https://dashboard.int.identitysandbox.gov).
+2. Choose [Create new test app](https://dashboard.int.identitysandbox.gov/service_providers/new).
+3. Enter the required fields to identify your service provider, including the **public** self-signed certificate you generated earlier.
 
-```ruby
-gem 'omniauth_login_dot_gov', :git => 'https://github.com/18f/omniauth_login_dot_gov.git'
-```
+#### Configure your app
 
-Now, configure the omniauth middleware and add it in an initializer:
-
-```ruby
-# config/initializers/omniauth.rb
-Rails.application.config.middleware.use OmniAuth::Builder do
-  provider :login_dot_gov, {
-    client_id: 'urn:gov:gsa:openidconnect:sp:myapp',
-    idp_base_url: 'http://localhost:3000/',
-    ial: 1,
-    private_key: OpenSSL::PKey::RSA.new(File.read('config/myapp.pem')),
-    redirect_uri: 'http://localhost:4040/auth/logindotgov/callback',
-  }
-end
-```
+Now that your app is configured with login.gov's sandbox as a Service Provider, you can begin configuring your local application.
+
+1. Copy your private key to your application's `config` directory (eg. `config/private.pem`)
+2. Add this gem to the Gemfile:
+  ```ruby
+  gem 'omniauth_login_dot_gov', :git => 'https://github.com/18f/omniauth_login_dot_gov.git'
+  ```
+3. Install this gem and dependencies with `bundle install`
+4. Now, configure the Omniauth middleware with an initializer:
+  ```ruby
+  # config/initializers/omniauth.rb
+  Rails.application.config.middleware.use OmniAuth::Builder do
+    provider :login_dot_gov, {
+      name: :login_dot_gov,
+      client_id: 'urn:gov:gsa:openidconnect:sp:myapp',
+      idp_base_url: 'https://idp.int.identitysandbox.gov/',
+      ial: 1,
+      private_key: OpenSSL::PKey::RSA.new(File.read('config/myapp.pem')),
+      redirect_uri: 'http://localhost:3000/auth/logindotgov/callback',
+    }
+  end
+  ```
```

Comment:
I feel like we need to provide some sample guidance on how to handle the callback, even if that’s just referencing omniauth docs.